pkg/providerid/providerid.go

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache v2.0 license.
+
+package providerid
+
+import (
+	"strings"
+
+	"github.com/microsoft/moc/pkg/errors"
+)
+
+const (
+	ProviderName     string = "moc"
+	ProviderIDPrefix string = ProviderName + "://"
+)
+
+type HostType string
```

Comment:
these enums can be defined in Proto, if required.

Reply:
@cwize1 FYI